src/PaymentGateways/PayPalCommerce/onBoardingRedirectHandler.php

```diff
@@ -274,37 +294,55 @@ private function didWeGetValidSellerRestApiCredentials( $array ) {
 	 * @param string $accessToken
 	 *
 	 * @param string $merchantId
+	 *
+	 * @return true|string[]
 	 */
 	private function isAdminSuccessfullyOnBoarded( $merchantId, $accessToken ) {
 		$onBoardedData = (array) $this->getSellerOnBoardingDetailsFromPayPal( $merchantId, $accessToken );
-		$required      = [ 'payments_receivable', 'primary_email_confirmed' ];
+		$required      = [ 'payments_receivable', 'primary_email_confirmed', 'products', 'capabilities' ];
 		$onBoardedData = array_filter( $onBoardedData ); // Remove empty values.
-		$errorMessage  = esc_html__( 'Your are successfully connected, but you need to do a few things within your PayPal account before you\'re ready to receive donations:', 'give' );
-		$redirect      = false;
+		$errorMessages = [];
 
 		if ( array_diff( $required, array_keys( $onBoardedData ) ) ) {
-			$this->redirectWhenOnBoardingFail();
+			return [ esc_html__( 'There was a problem with the status check for your account. Please try disconnecting and connecting again. If the problem persists, please contact support.', 'give' ) ];
 		}
 
 		if ( ! $onBoardedData['payments_receivable'] ) {
-			$redirect      = true;
-			$errorMessage .= sprintf(
-				'<br>- %1$s',
-				esc_html__( 'Set up an account to receive payment from PayPal', 'give' )
-			);
+			$errorMessages[] = esc_html__( 'Set up an account to receive payment from PayPal', 'give' );
 		}
 
 		if ( ! $onBoardedData['primary_email_confirmed'] ) {
-			$redirect      = true;
-			$errorMessage .= sprintf(
-				'<br>- %1$s',
-				esc_html__( 'Confirm your primary email address', 'give' )
-			);
+			$errorMessage[] = esc_html__( 'Confirm your primary email address', 'give' );
 		}
 
-		if ( $redirect ) {
-			$this->redirectWhenOnBoardingFail( $errorMessage );
+		// grab the PPCP_CUSTOM product from the status data
```

Comment:
Can we capitalize the first letter of comment?

Reply:
Resolved in 3471eb5